panel/_templates/pyodide_worker.js

```diff
@@ -13,29 +13,29 @@ async function startApplication() {
   self.postMessage({type: 'status', msg: 'Loading pyodide'})
   self.pyodide = await loadPyodide();
   self.pyodide.globals.set("sendPatch", sendPatch);
-  console.log("Loaded!");
+  console.log("Loaded pyodide!");
+  const data_archives = [{{ data_archives }}];
+  for (const archive of data_archives) {
+    let zipResponse = await fetch(archive);
+    let zipBinary = await zipResponse.arrayBuffer();
+    self.postMessage({type: 'status', msg: `Unpacking ${archive}`})
+    self.pyodide.unpackArchive(zipBinary, "zip");
+  }
   await self.pyodide.loadPackage("micropip");
-  const env_spec = [{{ env_spec }}]
-  for (const pkg of env_spec) {
-    let pkg_name;
-    if (pkg.endsWith('.whl')) {
-      pkg_name = pkg.split('/').slice(-1)[0].split('-')[0]
-    } else {
-      pkg_name = pkg
-    }
-    self.postMessage({type: 'status', msg: `Installing ${pkg_name}`})
-    try {
-      await self.pyodide.runPythonAsync(`
-        import micropip
-        await micropip.install('${pkg}');
-      `);
-    } catch(e) {
-      console.log(e)
-      self.postMessage({
-	type: 'status',
-	msg: `Error while installing ${pkg_name}`
-      });
-    }
+  self.postMessage({type: 'status', msg: `Installing packages`});
+  // a finegrained approach installing dependencies one after another with status updates was implemented previously
```

Comment:
Unfortunate but better to handle this correctly. Would be great if micropip supported hooks to report progress.

Reply:
ah, that was not commented properly for the review! The problem is that the dependency resolution fails for locally installed wheels (...) - probably this should be fixed in micropip. I will try to do some tests too and poke the micropip-people a bit in the meantime.

Reply:
Sweet thanks! Am trying to test this PR and then get it merged in time for the Panel 1.5 release.